packages/core/src/components/side-pane-group/side-pane-group.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Siemens AG
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import {
+  Component,
+  Element,
+  h,
+  Host,
+  Listen,
+  Prop,
+  State,
+} from '@stencil/core';
+
+const mobileBreakpoint: number = 767;
+
+@Component({
+  tag: 'ix-side-pane-group',
+  styleUrl: 'side-pane-group.scss',
+  shadow: true,
+})
+export class SidePaneGroup {
+  @Element() hostElement: HTMLIxSidePaneGroupElement;
+
+  /**
+   * Determines if the side panes behave inline
+   */
+  @Prop() inline: boolean = false;
+
+  /**
+   * Determines if the side panes behave floating
+   */
+  @Prop() floating: boolean = false;
+
+  /**
+   * Choose the variant of the panes
+   */
+  @Prop() variant: '1' | '2' = '2';
+
+  @State() mobileMode: boolean = false;
+
+  componentWillRender() {
+    if (this.inline && this.floating) {
+      console.error('Inline and floating can not be set at once!');
+      return;
+    }
+
+    this.onWindowResize();
+  }
+
+  componentDidRender() {
+    const sidePanels = this.getSidePanels();
+    sidePanels.forEach((sidePanelElement) => {
+      sidePanelElement.inline = this.inline;
+      sidePanelElement.floating = this.floating;
+    });
+
+    this.configureLayout();
+  }
+
+  configureLayout() {
+    const sidePanels = this.getSidePanels();
+    sidePanels.forEach((sidePanelElement) => {
+      let zIndex: string = '1';
+      const isBottomTop =
+        sidePanelElement.position === 'top' ||
+        sidePanelElement.position === 'bottom';
+
+      if (this.mobileMode) {
+        sidePanelElement.style.removeProperty('z-index');
+        return;
+      } else if (this.variant === '1') {
+        if (!isBottomTop) {
+          zIndex = '2';
+        }
+      } else {
+        if (isBottomTop) {
+          zIndex = '2';
+        }
+      }
+
+      sidePanelElement.style.zIndex = zIndex;
+    });
+  }
+
+  @Listen('resize', { target: 'window' })
+  onWindowResize() {
+    const newMode = window.innerWidth <= mobileBreakpoint;
+    if (this.mobileMode != newMode) {
+      this.mobileMode = newMode;
+    }
+  }
```

Comment:
Is there a reaspon not to use breakpoints utils (src/components/utils/breakpoints.ts) or [MediaQueryList](https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList)?